### PR TITLE
Removing the remote project dialog flag now everything is in place

### PR DESCRIPTION
**Pull request by AMZN-Phil**
Also use validation function now that the dependent PR has been merged

Signed-off-by: AMZN-Phil <pconroy@amazon.com>

### CODE REVIEW

File: Code/Tools/ProjectManager/Source/PythonBindingsInterface.h [hunk 1]
```diff
@@ -88,6 +88,15 @@ namespace O3DE::ProjectManager
...
 
 
+        // Remote source
+
+        /**
+         * Validates a repository without adding it.
+         * @param repoUri the absolute filesystem path or url to the repo.
+         * @return bool, true if the repository is valid
+         */
+        virtual bool ValidateRepository(const QString& repoUri) = 0;
```
Comment (lumberyard-employee-dm) [inline Code/Tools/ProjectManager/Source/PythonBindingsInterface.h:98]: This function should be marked `const`

**Review by AMZN-alexpete — APPROVED** [review]
+1 `ValidateRepository` function should be marked `const` if possible.

File: Code/Tools/ProjectManager/Source/PythonBindingsInterface.h (hunk 1 above)
Comment (AMZN-alexpete) [reply Code/Tools/ProjectManager/Source/PythonBindingsInterface.h:98]: +1

Comment (AMZN-Phil) [reply Code/Tools/ProjectManager/Source/PythonBindingsInterface.h:98]: This function calls ExecuteWithLock which is not const due to clearing/possibly writing to local copies of error strings returned from Python.

**Review by lumberyard-employee-dm — APPROVED** [review]

File: Code/Tools/ProjectManager/Source/PythonBindingsInterface.h (hunk 1 above)
Comment (lumberyard-employee-dm) [reply Code/Tools/ProjectManager/Source/PythonBindingsInterface.h:98]: Shouldn't ExecuteWithLock be marked `const` with the PythonBindings::m_lock mutex marked `mutable`?
Semantically ExecuteWithLock is const function as it doesn't modify the state of the python bindings.

**Pull request merged**